pandas/core/internals/managers.py

```diff
@@ -362,7 +363,22 @@ def shift(self: T, periods: int, axis: int, fill_value) -> T:
         if fill_value is lib.no_default:
             fill_value = None
 
-        if axis == 0 and self.ndim == 2 and self.nblocks > 1:
+        if (
+            axis == 0
```

Comment:
this is super complicated can you split into multiple clauses & add comments

Reply:
sure will comment.  fortunately we can rip out some of this in 2.0

Reply:
comment added + green